### PR TITLE
fix(store): enforce namespace isolation

### LINKED ISSUE #275: [BUG] Store Namespace Isolation Bypass — Any User Can Read/Write Any User's Data
From recent run of [Thresher](https://github.com/thresher-sh/thresher) against repo, surfaced bug on store. Likely forgotten todo.

### CRIT-002 — Store Namespace Isolation Bypass — Any User Can Read/Write Any User's Data

| Field | Value |
|-------|-------|
| **Title** | Development backdoor in `apply_user_namespace_scoping()` never removed |
| **Source** | AI Analyst: The Behaviorist (Unknown CVE Hunter) |
| **File** | `libs/aegra-api/src/aegra_api/api/store.py` |
| **Lines** | 236–248 |
| **CVE / CWE** | CWE-284 (Improper Access Control), CWE-732 (Incorrect Permission Assignment) |
| **CVSS** | N/A (no assigned CVE; estimated ~9.1 AV:N/AC:L/PR:L/UI:N/S:U/C:H/I:H/A:N) |
| **EPSS** | N/A |
| **CISA KEV** | No |
| **AI Risk Score** | CRITICAL, 98% confidence |
| **Adversarial Status** | Not adversarially verified |

**Description:**  
The function `apply_user_namespace_scoping()` is the sole data-isolation mechanism for the key-value store. Its logic is:

1. If no namespace is provided, scope to `["users", user_id]` — correct.
2. If the namespace starts with `["users", user_id]`, pass it through — correct.
3. **Otherwise: return the caller-supplied namespace unmodified.** — This is the bug.

Branch 3 is the catch-all for every namespace that does NOT belong to the calling user. Rather than denying the request or remapping to the caller's namespace, it silently passes through the attacker-controlled namespace. An in-code comment admits the intent: `# For development, allow all namespaces (remove this for production)` — but the guard was never removed.

Any authenticated user can send `PUT /store/items` with `{"namespace": ["users", "victim_id"], "key": "secret", "value": {...}}` and overwrite a victim's data. The same path exists for GET (exfiltration), DELETE (data destruction), and SEARCH (enumeration of all users' data).

**Evidence:**
```python
def apply_user_namespace_scoping(user_id: str, namespace: list[str]) -> list[str]:
    if not namespace:
        return ["users", user_id]
    if namespace[0] == "users" and len(namespace) >= 2 and namespace[1] == user_id:
        return namespace
    # For development, allow all namespaces (remove this for production)
    return namespace   # <-- BUG: returns attacker-supplied namespace
```

**Remediation:**  
Replace the final `return namespace` with a deny (raise an authorization error) or a scoped prefix: `return ["users", user_id] + namespace[2:]`. Remove the development comment. Add integration tests that verify cross-user namespace access is rejected with a 403.

### PR DESCRIPTION
**Pull request by ibbybuilds**
## Summary
- **Security fix:** `apply_user_namespace_scoping()` had a development catch-all that returned attacker-supplied namespaces unmodified, allowing any authenticated user to read/write/delete other users' store data
- Unrecognized namespaces are now scoped under the caller's `["users", user_id]` prefix instead of being passed through
- Added 10 unit tests covering isolation enforcement (cross-user, arbitrary namespace, edge cases)
- Bumped version 0.8.4 → 0.8.5

## What changed
| File | Change |
|------|--------|
| `api/store.py` | Replaced `return namespace` catch-all with `return ["users", user_id] + namespace` |
| `tests/unit/test_api/test_store_namespace_scoping.py` | New — 10 tests for namespace isolation |
| Both `pyproject.toml` | 0.8.4 → 0.8.5 |

## Test plan
- [x] 10/10 new unit tests pass
- [x] 33/33 existing store integration tests pass
- [x] Lint clean

Closes #275

🤖 Generated with [Claude Code](https://claude.com/claude-code)

## Summary by CodeRabbit

* **Bug Fixes**
  * Enforced stricter namespace scoping for store operations so requests are consistently scoped to the authenticated user, preventing cross-user access.

* **Documentation**
  * Updated store namespace scoping guidance to document the new authenticated-user scoping behavior.

* **Tests**
  * Added unit and integration tests covering namespace remapping for store CRUD and search operations.

* **Chores**
  * Bumped project versions to 0.8.5.

### CODE REVIEW

File: libs/aegra-api/tests/unit/test_api/test_store_namespace_scoping.py [hunk 1]
```diff
@@ -0,0 +1,61 @@
+"""Tests for store namespace scoping / user isolation."""
+
+from aegra_api.api.store import apply_user_namespace_scoping
```
[the remark this answers is not in the document]
Comment (ibbybuilds) [reply libs/aegra-api/tests/unit/test_api/test_store_namespace_scoping.py:3]: Added integration tests for all four cross-user namespace paths (PUT, GET, DELETE, SEARCH). Each asserts the store receives the remapped namespace tuple.

File: libs/aegra-api/pyproject.toml [hunk 2]
```diff
@@ -1,6 +1,6 @@
 [project]
 name = "aegra-api"
-version = "0.8.4"
+version = "0.8.5"
```
[the remark this answers is not in the document]
Comment (ibbybuilds) [reply libs/aegra-api/pyproject.toml:3]: Patch bump is intentional. The old behavior was a development backdoor explicitly marked "For development, allow all namespaces (remove this for production)" — no user should have been relying on it. The fix restores the originally intended behavior, which makes it a bug fix. Will call it out in release notes for anyone who was unknowingly depending on the pass-through.

File: libs/aegra-api/src/aegra_api/api/store.py [hunk 3]
```diff
@@ -234,15 +234,16 @@ async def list_namespaces(
...
         return ["users", user_id]
 
-    # Allow explicit user namespaces
     if namespace[0] == "users" and len(namespace) >= 2 and namespace[1] == user_id:
         return namespace
 
-    # For development, allow all namespaces (remove this for production)
-    return namespace
+    # Scope any other namespace under the user's prefix
+    return ["users", user_id] + namespace
```
[the remark this answers is not in the document]
Comment (ibbybuilds) [reply libs/aegra-api/src/aegra_api/api/store.py:249]: Added namespace scoping documentation to docs/guides/store.mdx — explains the three scoping rules (empty -> default, own prefix -> passthrough, anything else -> prefixed under user).

File: libs/aegra-api/tests/integration/test_api/test_store_crud.py [hunk 4]
```diff
@@ -595,6 +595,65 @@ def test_get_item_applies_user_scoping(self, client, mock_store):
...
+            json={
+                "namespace_prefix": ["users", "other-user", "docs"],
+            },
+        )
+
+        assert resp.status_code == 200
+        call_args = mock_store.asearch.call_args
+        namespace_prefix = call_args[0][0]
+        assert namespace_prefix == ("users", "test-user", "users", "other-user", "docs")
+
```
[the remark this answers is not in the document]
Comment (ibbybuilds) [reply libs/aegra-api/tests/integration/test_api/test_store_crud.py:656]: Won't fix — none of the 30+ existing test methods in this file have parameter annotations. Adding them only to our 4 new methods would be inconsistent. If we want to enforce this in tests, it should be a separate cleanup pass across the whole test suite.
